src/internal/events/events.go

```diff
@@ -165,8 +169,15 @@ func storageID(s storage.Storage) string {
 	return id
 }
 
-func repoHash(s storage.Storage) string {
-	return md5HashOf(storageID(s))
+// NewRepoID generates a new unique repository id hash.
+// Repo ids should only be generated once per repository,
+// and must be stored after that.
+func NewRepoID(s storage.Storage) string {
+	return repoHash(s, uuid.NewString())
+}
```

Comment:
this should be moved to another package. Events has nothing to do with generating and storing a repo ID

Reply:
Well, I wouldn't say nothing, since it's the only thing that uses the repo ID.  But I agree on the package move.